### PR TITLE
RealTime GPS Data logger project

### PR DESCRIPTION
**Pull request by Ghost-Ashu**
## Describe your project in 1-3 sentences
i will be building a realtime GPS Data Logger by using Google Maps API and NodeMCU ESP8266. It will Display the geographic location of a user or device on a Google map

## Hack Club Slack handle:
ashutoshkumarsingh

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Comment by cfanoulis** [comment]
Hey, this sounds very nice! real quick, can you please include your slack username in the PR description? thaanks!

**Comment by Ghost-Ashu** [comment]
> Hey, this sounds very nice! real quick, can you please include your slack username in the PR description? thaanks!

Thank You so much i have updated my Slack username. 😊

**Review by Ghost-Ashu — COMMENTED** [review]
reviwed

File: Ghost-Ashu.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: "@Ghost-Ashu"
+project: "RealTime GPS Data Logger with GMaps"
+---
+---
+#### Email: ashutoshkumarsingh951@gmail.com
```
[the remark this answers is not in the document]
Comment (Ghost-Ashu) [reply Ghost-Ashu.md]: Sorry! for including my email, i have removed it.

File: Ghost-Ashu.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Summary
+
+Have You ever wonder! How Gobal Positioning System Tracks and Logs our Cordinates. Here  is my Project Walkthrough, where i will be building a realtime GPS Data Logger by using  Google Maps API and NodeMCU ESP8266.
+
+## Technologies Used
+Aurdino (IDE) software
+Google Cloud Platform (GCP) for using google Maps API
+
+## Plan
```
[the remark this answers is not in the document]
Comment (Ghost-Ashu) [reply Ghost-Ashu.md:16]: i Have worked with Network based Iot Devices in my past. Specially! the GPS modules, so with the Circuit, i would not face any issues while building this project.

File: Ghost-Ashu.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+--> Arduino Uno Rev3 can be used in my project, as it is a microcontroller board based on the ATmega328 microcontroller. One way it could be used to collect GPS data from a GPS module, process the data to extract the desired information (e.g. latitude, longitude, altitude), and then send the data to a remote server or cloud platform over a wired or wireless connection. To use the Arduino Uno in this project, you would need to connect it to the GPS module using a serial communication interface such as UART or I2C.
+-->for  Power Supply i will use an external power supply, then connect it to the barrel jack or VIN pin on the Arduino Uno board. The barrel jack can accept a voltage range of 7-12V DC. while the VIN pin can accept a regulated 5V or 6V DC. also can use power source (e.g. battery or AC adapter).
+--> i already have my casing for my project. and Necessary other requirements. This is my Dream Project as this would my first GPS module Project that i have worked in years. i would be happy if u agree to approve my Project.
```
[the remark this answers is not in the document]
Comment (Ghost-Ashu) [reply Ghost-Ashu.md:22]: Casing for my Project i Have added it :)

File: Ghost-Ashu.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+| -------------------------------------------------| --------------------------------------------------------------------------------| ------ |
+| Arduino Uno Rev3                                 |     https://store.arduino.cc/products/arduino-uno-rev3                          | $25.60 |
+| NodeMCU ESP8266 (AMICA-CP2102 driver)            |     https://robocraze.com/products/nodemcu-esp8266-amica-cp2102-driver          | $4.10  |
+|BN-282 GPS Module(NMEA-0183)                      |     https://bit.ly/3WxojpM                                                      | $53.71 |
+|3.5 inch TFT SPI LCD Display Resolution 480 x 320 |     https://amzn.eu/d/3HccX7w						                                       | $64.55 |
+|Jumper Wires					                             |     https://a.co/d/aHmKe1U							                                         | $6.98  |
+|Soldering Iron Kit                                |     https://robu.in/product/plusivo-soldering-kit-for-electronics               | $12.66 |
+| Total                                            |                                                                                 | $167.6 |
```
[the remark this answers is not in the document]
Comment (Ghost-Ashu) [reply Ghost-Ashu.md:39]: yes! all the Products i have mentioned here are including all the Necessary Taxes and Shipping Charges keeping in mind.

**Review by cfanoulis — CHANGES_REQUESTED** [review]
Thanks for that. I can't understand what you'll be displaying on your screen - will it be an image from google maps?

Also, how will the NodeMCU & arduino be connected? Esp. considering the arduino device doesn't have WiFi capabilities - are both even required?

**Comment by cfanoulis** [comment]
Hello Ghost! It has been a while since my last message - and we need to know if you're still interested in a winter grant. If you are - please reply to this PR (or this email) within the next 48 hours - else we will be closing this proposal. Thanks :)

**Comment by Ghost-Ashu** [comment]
> Thanks for that. I can't understand what you'll be displaying on your screen - will it be an image from google maps?
> 
> Also, how will the NodeMCU & arduino be connected? Esp. considering the arduino device doesn't have WiFi capabilities - are both even required?
let me Explain u again,  GPS module in my project is  to determine the device's current location. The GPS module receives signals from a network of satellites orbiting the earth, and uses this information to calculate the device's latitude, longitude, and altitude.
The NodeMCU can act as the main controller, while the Arduino can handle the GPS module. The NodeMCU can send commands to the Arduino to retrieve the GPS data, and the Arduino can send the data back to the NodeMCU to be processed and stored.
🙁 i guess u got. i have also included all these details in my PR.
still from many days i keep changing the details according to your Guidelines.

**Comment by Ghost-Ashu** [comment]
> Hello Ghost! It has been a while since my last message - and we need to know if you're still interested in a winter grant. If you are - please reply to this PR (or this email) within the next 48 hours - else we will be closing this proposal. Thanks :)

Sorry! for not replying due to my college  exams. i will be Thankful if my proposal gets accepted😊

**Comment by cfanoulis** [comment]
hey - sorry for making too many questions - i really need to understand how your project works before I give the a-ok hah.

How does Google Maps tie into this? Will you be displaying location information on your display? or would you have a seperate website where the location would be shown

**Comment by Ghost-Ashu** [comment]
> 

The location would be Displayed on my TFT screen. Where ever i go. The GPS module provides only GPS coordinates, which would not be much accurate or presise.  so I would be using an GCP ( google cloud platform) maps API for providing more accurate results from those coordinates.

**Review by cfanoulis — APPROVED** [review]
Great! Thank you so much for bearing with me here :)

A banker will be in touch (after the weekend) to get you setup on Bank. Enjoy the grant (responsibly)!

**Comment by Ghost-Ashu** [comment]
> Great! Thank you so much for bearing with me here :)
> 
> A banker will be in touch (after the weekend) to get you setup on Bank. Enjoy the grant (responsibly)!

Thank you so much ❤️

**Review by MelSmith104 — COMMENTED** [review]
hey @Ghost-Ashu! this project sounds super interesting :) im dm'ing you on slack to chat about next steps for the grant!

**Comment by MelSmith104** [comment]
closing pr due to unanswered questions on Slack

**Pull request closed without merging**